newIDE/app/src/MainFrame/Preferences/PreferencesContext.js

```diff
@@ -168,6 +170,10 @@ export type Preferences = {|
   insertRecentProjectFile: (
     fileMetadata: FileMetadataAndStorageProviderName
   ) => void,
+  getAutoOpenMostRecent: () => boolean,
+  setAutoOpenMostRecent: (enabled: boolean) => void,
+  getProjectHasOpened: () => boolean,
+  setProjectHasOpened: (enabled: boolean) => void,
```

Comment:
nit: let's call this `hadProjectOpenedDuringLastSession` and `setHasProjectOpened`